src/Nethermind/Nethermind.Core2/IEnumerableExtensions.cs

```diff
@@ -14,12 +14,21 @@
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
 
+using System.Collections.Generic;
+using System.Linq;
+
 namespace Nethermind.Core2
 {
-    /// <summary>
-    /// Temporarily - constants that are not assigned anywhere...
-    /// </summary>
-    public static class Constants
+    public static class IEnumerableExtensions
     {
+        public static IList<T> AsList<T>(this IEnumerable<T> enumerable)
```

Comment:
Should be called AsIList, as it returns and IList, not List.

Not sure of the value of this as an extension method. Seems like more confusion added to the existing Linq methods, as it is like them but not quite. 

If you are receiving something and want to index it, then accept IList.

There are also different semantics, as if it happens to be IList then you return the reference, which is mutable, whereas if it is not, then you copy the elements into a new list, which is not mutable, i.e. if you add things to it later it doesn't change.

This is a pretty subtle change by changing the input whether it takes a reference or copies.

Reply:
IEnumerable can be a nightmare if that is multithreaded - better always take snapshots. Just added once so you can have a look and bear in mind but can leave IEnumerable for now everywhere else.